# Homework 7
## Problem 4
Estimate flowrate in $\frac{kg}{hr}$ for various solvents when given the feed flowrate, compositions of feed and raffinate streams, and partition coefficients

Starting with a mass balance on each component,
$$
\omega_A^FF=\omega_A^RR \tag{1}
$$
$$
\omega_B^FF=\omega_B^RR+\omega_B^EE \tag{2}
$$
$$
S = \omega_C^EE \tag{3}
$$
([1](#mjx-eqn-1)), ([2](#mjx-eqn-2)), and ([3](#mjx-eqn-3)) can be combined to get,
$$
W_B^FF_A=W_B^ES+W_B^RF_A \tag{4}
$$
where $W_B^i = \frac{\omega_B^i}{\omega_A^i}$ and $F_A = \omega_A^FF$

The partition coefficient can be found by
$$
K_D = \frac{\omega_B^E}{\omega_B^R} \tag{5}
$$
or
$$
K_D' = \frac{W_B^E}{W_B^R} = K_D\frac{1-\omega_B^R}{1-\omega_B^E} \tag{6}
$$

([4](#mjx-eqn-4)) and ([6](#mjx-eqn-6)) can be combined to get
$$
\frac{W_B^R}{W_B^F} = \frac{1}{(1+\frac{K_D'S}{F_A})}
$$
which can be rearranged to get
$$
S = \frac{(\frac{W_B^F}{W_B^R}-1)F_A}{K_D'} \tag{7}
$$
To find $K_D'$, $\omega_B^E$ can be found using ([5](#mjx-eqn-5))

In [16]:
import numpy as np

In [17]:
wBF = 0.08                                  #mass frac B in feed
wAF = 1-wBF                                 #mass frac A in feed
wBR = 0.01                                  #mass frac B in raff
wAR = 1-wBR                                 #mass frac A in raff

F = 13500                                   #feed flowrate kg/hr
FA = F*wAF                                  #flowrate of A in feed

KD = np.array([1.273,.429,.312,.178])       #partitioin coeffs (mass frac)

wBE = KD*wBR                                #mass frac B in extract

KDp = KD*(1-wBR)/(1-wBE)                    #partition coeff (mass ratio)

In [18]:
def S(wbF,waF,wbR,waR,FA,KDp):
    XbF = wbF/waF
    XbR = wbR/waR
    S = (XbF/XbR-1)*FA/KDp
    return S
print(S(wBF,wAF,wBR,wAR,FA,KDp))

[ 74029.38656002 221550.22250477 304989.51048951 535306.94586313]


|Solvent|S (kg/hr)|
|-----|-----|
|Methyl Acetate|74,029|
|Isopropyl Ether|221,550|
|Heptadecanol|304,990|
|Chloroform|535,307|